src/components/Question.tsx

```diff
@@ -0,0 +1,350 @@
+      }
+    }
+  }
+
+  .openstax-answer {
+    border-top: 1px solid #d5d5d5;
+    margin: 10px 0;
+    padding: 10px 0;
+  }
+}
+`;
+
+export interface QuestionProps {
+  question: QuestionType,
+  answer_id?: string;
+  task?: {
+    is_deleted: boolean;
+    type?: 'homework'
+  } | null,
+  correct_answer_id: ID;
+  incorrectAnswerId: ID;
+  hideAnswers: boolean;
+  hidePreambles: boolean,
+  exercise_uid?: string;
+  displayFormats: boolean,
+  processHtmlAndMath: boolean;
+  className: string;
+  questionNumber: number;
+  displaySolution: boolean;
+  context?: string;
+  correct_answer_feedback_html?: string;
+  contentRenderer?: JSX.Element;
+  feedback_html: string;
+  onChange: () => void;
+  show_all_feedback?: boolean;
+  children?: ReactNode;
+}
+
+export const Question = (props: QuestionProps) => {
+  let exerciseUid, solution;
+
+  const {
+    question, correct_answer_id, incorrectAnswerId, exercise_uid, className, questionNumber, context, task, hidePreambles
+  } = props;
+
+  const { stem_html, collaborator_solutions, formats, stimulus_html } = question;
+
+  const hasCorrectAnswer = !!correct_answer_id;
+  const hasIncorrectAnswer = !!incorrectAnswerId;
+
+  const taskIsDeleted = (task != null ? task.is_deleted : undefined)
+  const taskIsHomework = ((task != null ? task.type : undefined) === 'homework');
+
+  const classes = classnames('openstax-question', className, {
+    'has-correct-answer': hasCorrectAnswer && !(taskIsDeleted && taskIsHomework),
+    'has-incorrect-answer': hasIncorrectAnswer,
+  });
+
+  const hasSolution = () => {
+    const { question, displaySolution } = props;
+    const { collaborator_solutions } = question;
+
+    return (
+      displaySolution && collaborator_solutions.find(s => s['content_html'] !== undefined)
+    );
+  };
+
+  if (exercise_uid != null) {
+    exerciseUid = (
+      <div className="exercise-uid">
+        {exercise_uid}
+      </div>
+    );
+  }
+
+  if (hasSolution()) {
+    solution =
+      <div className="detailed-solution">
+        <div className="header">
+          Detailed solution:
+        </div>
+        <Content
+          className="solution"
+          block={true}
+          html={collaborator_solutions.map(s => s['content_html']).join('')} />
+      </div>;
+  }
+
+  return (
+    <StyledQuestion className={classes} data-question-number={questionNumber} data-test-id="question">
+      <QuestionHtml type="context" html={context} hidden={hidePreambles} />
+      <QuestionHtml type="stimulus" html={stimulus_html} hidden={hidePreambles} />
+      <QuestionHtml type="stem" html={stem_html} hidden={hidePreambles} questionNumber={questionNumber} />
+      {props.children}
+
+      <AnswersTable
+        {...props}
+        onChangeAnswer={props.onChange}
+        hasCorrectAnswer={hasCorrectAnswer} />
+
+      {solution}
+      {props.displayFormats ? <FormatsListing formats={formats} /> : undefined}
```

Comment:
is this something that will happen outside of tutor? if not maybe we could just call it something generic and allow any component to be passed in?